pkg/model/resource/options.go

```diff
@@ -0,0 +1,259 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/gobuffalo/flect"
+
+	"sigs.k8s.io/kubebuilder/pkg/model/config"
+)
+
+const (
+	versionPattern = "^v\\d+(alpha\\d+|beta\\d+)?$"
+
+	groupRequired   = "group cannot be empty"
+	versionRequired = "version cannot be empty"
+	kindRequired    = "kind cannot be empty"
+)
+
+var (
+	versionRegex = regexp.MustCompile(versionPattern)
+
+	v1CoreGroups = map[string]string{
+		"apps":                  "",
+		"admissionregistration": "k8s.io",
+		"apiextensions":         "k8s.io",
+		"authentication":        "k8s.io",
+		"autoscaling":           "",
+		"batch":                 "",
+		"certificates":          "k8s.io",
+		"core":                  "",
+		"extensions":            "",
+		"metrics":               "k8s.io",
+		"policy":                "",
+		"rbac.authorization":    "k8s.io",
+		"storage":               "k8s.io",
+	}
+
+	v2CoreGroups = map[string]string{
+		"admission":             "k8s.io",
+		"admissionregistration": "k8s.io",
+		"apps":                  "",
+		"auditregistration":     "k8s.io",
+		"apiextensions":         "k8s.io",
+		"authentication":        "k8s.io",
+		"authorization":         "k8s.io",
+		"autoscaling":           "",
+		"batch":                 "",
+		"certificates":          "k8s.io",
+		"coordination":          "k8s.io",
+		"core":                  "",
+		"events":                "k8s.io",
+		"extensions":            "",
+		"imagepolicy":           "k8s.io",
+		"networking":            "k8s.io",
+		"node":                  "k8s.io",
+		"metrics":               "k8s.io",
+		"policy":                "",
+		"rbac.authorization":    "k8s.io",
+		"scheduling":            "k8s.io",
+		"setting":               "k8s.io",
+		"storage":               "k8s.io",
+	}
+)
+
+// Options contains the information required to build a new Resource
+type Options struct {
+	// Group is the API Group. Does not contain the domain.
+	Group string
+
+	// Version is the API version.
+	Version string
+
+	// Kind is the API Kind.
+	Kind string
+
+	// Plural is the API Kind plural form.
+	// Optional
+	Plural string
+
+	// Namespaced is true if the resource is namespaced.
+	Namespaced bool
+
+	// CreateExampleReconcileBody will create a Deployment in the Reconcile example.
+	// v1 only
+	CreateExampleReconcileBody bool
+}
+
+func (opts *Options) Validate() error {
+	// Check that the required flags did not get a flag as their value
```

Comment:
I remember that we tried to work around this issue by checking if it starts with `-` last year.
I wonder if the issue has been fix by upstream Cobra. If so, we can just bump the Cobra version.

Reply:
Cobra stated that this is normal unix-like flag parsing and is not a bug, so not luck in that sense.